### PR TITLE
tso: implement groupSplitPatroller to speed up the split process

### LINKED ISSUE #6696: Split state cannot recover if there is no request
## Enhancement
If the split target group doesn't receive the request, it will stay in the split state, which cannot be selected as the split source.

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #5895, close #6696.

### What is changed and how does it work?

```commit-message
Implement `groupSplitPatroller` to speed up the split process.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

**Comment by lhy1024** [comment]
IMO, it will close https://github.com/tikv/pd/issues/6696?

File: pkg/tso/keyspace_group_manager.go [hunk 1]
```diff
@@ -730,6 +735,10 @@ func (kgm *KeyspaceGroupManager) updateKeyspaceGroup(group *endpoint.KeyspaceGro
 	}
 	kgm.kgs[group.ID] = group
 	kgm.ams[group.ID] = am
+	// If the group is the split target, add it to the splitting group map.
+	if group.IsSplitTarget() {
+		kgm.splittingGroups[group.ID] = struct{}{}
```
Comment (rleungx) [inline pkg/tso/keyspace_group_manager.go:740]: What if it transfers the leader? Will splittingGroups become invalidated?

Comment (JmPotato) [reply pkg/tso/keyspace_group_manager.go:740]: `groupSplitPatroller` will check if the current allocator is the leader.

**Review by rleungx — APPROVED** [review]

**Pull request merged**